core/src/main/java/io/questdb/griffin/engine/functions/date/ToTimezoneIntervalFunctionFactory.java

```diff
@@ -0,0 +1,143 @@
+            return new ToTimezoneIntervalFunctionVar(interval, timezone);
+        }
+    }
+
+    @NotNull
+    static IntervalFunction getIntervalFunction(IntList argPositions, Function interval, Function timezone, int multiplier) throws SqlException {
+        final CharSequence tz = timezone.getStrA(null);
+        if (tz != null) {
+            final int hi = tz.length();
+            final long l = Timestamps.parseOffset(tz, 0, hi);
+            if (l == Long.MIN_VALUE) {
+                try {
+                    return new OffsetIntervalFunctionFromRules(
+                            interval,
+                            TimestampFormatUtils.EN_LOCALE.getZoneRules(
+                                    Numbers.decodeLowInt(TimestampFormatUtils.EN_LOCALE.matchZone(tz, 0, hi)), RESOLUTION_MICROS
+                            ),
+                            multiplier
+                    );
+                } catch (NumericException e) {
+                    Misc.free(interval);
+                    throw SqlException.$(argPositions.getQuick(1), "invalid timezone name");
+                }
+            } else {
+                return new OffsetIntervalFunctionFromOffset(
+                        interval,
+                        multiplier * Numbers.decodeLowInt(l) * Timestamps.MINUTE_MICROS
+                );
+            }
+        }
+        throw SqlException.$(argPositions.getQuick(1), "timezone must not be null");
+    }
+
+    private static class ToTimezoneIntervalFunctionVar extends IntervalFunction implements BinaryFunction {
+        private final Function interval;
+        private final Function timezone;
+
+        public ToTimezoneIntervalFunctionVar(Function interval, Function timezone) {
+            this.interval = interval;
+            this.timezone = timezone;
+        }
+
+        @Override
+        public Interval getInterval(Record rec) {
+            final Interval localInterval = new Interval(interval.getInterval(rec).getLo(), interval.getInterval(rec).getHi());
+            try {
+                final CharSequence tz = timezone.getStrA(rec);
+                if (tz != null) {
+                    localInterval.of(
+                            Timestamps.toTimezone(localInterval.getLo(), TimestampFormatUtils.EN_LOCALE, tz),
+                            Timestamps.toTimezone(localInterval.getHi(), TimestampFormatUtils.EN_LOCALE, tz)
+                    );
+                    return localInterval;
+                } else {
+                    return localInterval;
+                }
+            } catch (NumericException e) {
+                return localInterval;
+            }
+        }
+
+        @Override
+        public Function getLeft() {
+            return interval;
+        }
+
+        @Override
+        public String getName() {
+            return "to_timezone";
+        }
+
+        @Override
+        public Function getRight() {
+            return timezone;
+        }
+    }
+}
```

Comment:
please add `toPlan(Sink)` function to serialize argument values and add test for plan (e.g. `explain select ...`)